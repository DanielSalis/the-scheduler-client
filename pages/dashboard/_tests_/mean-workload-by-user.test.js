import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import MeanWorkloadByUser from '../mean-workload-by-user.vue'
import VueApexCharts from 'vue-apexcharts'

// Mock the layout
const MockLayout = {
  template: '<div><slot /></div>',
}

// Mock VueApexCharts
const MockVueApexCharts = {
  template: '<div></div>',
}

describe('MeanWorkloadByUser.vue', () => {
  let localVue
  let vuetify

  const wrapper = shallowMount(MeanWorkloadByUser, {
    localVue,
    vuetify,
    data() {
      return {
        chart: true,
        chartOptions: { "chart": { "height": 350, "type": "line", "dropShadow": { "enabled": true, "color": "#000", "top": 18, "left": 7, "blur": 10, "opacity": 0.2 }, "toolbar": { "show": true } }, "colors": ["#77B6EA", "#545454"], "dataLabels": { "enabled": true }, "stroke": { "curve": "smooth" }, "title": {}, "markers": { "size": 1 }, "xaxis": {}, "yaxis": { "title": { "text": "Quantidade de minutos" }, "labels": { "show": true, "formatter": { "_custom": { "type": "function", "display": "<span style=\"opacity:.5;\">function</span> formatter(val)", "tooltip": "<pre>function formatter(val) {\n              return val + \" minutos\";\n            }</pre>", "_reviveId": 11 } } } }, "legend": { "position": "top", "horizontalAlign": "right", "floating": true, "offsetY": -25, "offsetX": -5 }, "theme": { "mode": "dark", "palette": "palette1", "monochrome": { "enabled": false, "color": "#255aee", "shadeTo": "light", "shadeIntensity": 0.65 } } },
        charts: [{ "title": "Milena Matos", "email": "milena@email.com", "workload": 165, "series": [{ "name": "Quantidade de minutos", "data": [165] }], "chartOptions": { "chart": { "height": 350, "type": "line", "dropShadow": { "enabled": true, "color": "#000", "top": 18, "left": 7, "blur": 10, "opacity": 0.2 }, "toolbar": { "show": true } }, "colors": ["#77B6EA", "#545454"], "dataLabels": { "enabled": true }, "stroke": { "curve": "smooth" }, "title": { "text": "Quantidade de minutos trabalhados por dia - Milena Matos", "align": "left" }, "markers": { "size": 1 }, "xaxis": { "categories": ["2024-08-08"], "title": { "text": "Dias" } }, "yaxis": { "title": { "text": "Quantidade de minutos" }, "labels": { "show": true, "formatter": { "_custom": { "type": "function", "display": "<span style=\"opacity:.5;\">function</span> formatter(val)", "tooltip": "<pre>function formatter(val) {\n              return val + \" minutos\";\n            }</pre>", "_reviveId": 7 } } } }, "legend": { "position": "top", "horizontalAlign": "right", "floating": true, "offsetY": -25, "offsetX": -5 }, "theme": { "mode": "dark", "palette": "palette1", "monochrome": { "enabled": false, "color": "#255aee", "shadeTo": "light", "shadeIntensity": 0.65 } }, "labels": ["2024-08-08"] } }, { "title": "daniel", "email": "daniel@mail.com", "workload": 115, "series": [{ "name": "Quantidade de minutos", "data": [115] }], "chartOptions": { "chart": { "height": 350, "type": "line", "dropShadow": { "enabled": true, "color": "#000", "top": 18, "left": 7, "blur": 10, "opacity": 0.2 }, "toolbar": { "show": true } }, "colors": ["#77B6EA", "#545454"], "dataLabels": { "enabled": true }, "stroke": { "curve": "smooth" }, "title": { "text": "Quantidade de minutos trabalhados por dia - daniel", "align": "left" }, "markers": { "size": 1 }, "xaxis": { "categories": ["2024-08-08"], "title": { "text": "Dias" } }, "yaxis": { "title": { "text": "Quantidade de minutos" }, "labels": { "show": true, "formatter": { "_custom": { "type": "function", "display": "<span style=\"opacity:.5;\">function</span> formatter(val)", "tooltip": "<pre>function formatter(val) {\n              return val + \" minutos\";\n            }</pre>", "_reviveId": 8 } } } }, "legend": { "position": "top", "horizontalAlign": "right", "floating": true, "offsetY": -25, "offsetX": -5 }, "theme": { "mode": "dark", "palette": "palette1", "monochrome": { "enabled": false, "color": "#255aee", "shadeTo": "light", "shadeIntensity": 0.65 } }, "labels": ["2024-08-08"] } }],
        filter: { "start": "2024-08-08", "end": null },
        generalChart: { "series": [{ "name": "Minutos trabalhados", "data": [165, 115] }], "chartOptions": { "chart": { "height": 350, "type": "bar" }, "plotOptions": { "bar": { "borderRadius": 10, "dataLabels": { "position": "top" } } }, "dataLabels": { "enabled": true, "formatter": { "_custom": { "type": "function", "display": "<span style=\"opacity:.5;\">function</span> formatter(val)", "tooltip": "<pre>function formatter(val) {\n                        return val;\n                      }</pre>", "_reviveId": 9 } }, "offsetY": -20, "style": { "fontSize": "12px" } }, "xaxis": { "categories": ["Milena Matos", "daniel"], "position": "bottom", "axisBorder": { "show": true }, "axisTicks": { "show": false }, "crosshairs": { "fill": { "type": "gradient", "gradient": { "colorFrom": "#D8E3F0", "colorTo": "#BED1E6", "stops": [0, 100], "opacityFrom": 0.4, "opacityTo": 0.5 } } }, "tooltip": { "enabled": true }, "convertedCatToNumeric": false }, "yaxis": { "axisBorder": { "show": false }, "axisTicks": { "show": false }, "labels": { "show": true, "formatter": { "_custom": { "type": "function", "display": "<span style=\"opacity:.5;\">function</span> formatter(val)", "tooltip": "<pre>function formatter(val) {\n                          return val;\n                        }</pre>", "_reviveId": 10 } } } }, "title": { "text": "Funcionários no dia 2024-08-08", "align": "center" }, "theme": { "mode": "dark", "palette": "palette3", "monochrome": { "enabled": false, "color": "#255aee", "shadeTo": "light", "shadeIntensity": 0.75 } } } },
        menuEnd: false,
        menuStart: false,
        panel: [],
        series: null,
        userFilter: "",
      }
    },
    mocks: {
      $axios: {
        get: jest.fn(() => Promise.resolve({ data: [] })),
      },
    },
    stubs: {
      layout: MockLayout,
      VueApexCharts: MockVueApexCharts,
    },
  })

  beforeEach(() => {
    localVue = createLocalVue()
    vuetify = new Vuetify()
    localVue.use(Vuetify)
  })

  it('renders the component and matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders v-components', () => {
    expect(wrapper.findAllComponents({ name: 'v-date-picker' }).length).toBe(2)
    expect(wrapper.findAllComponents({ name: 'v-expansion-panel' }).length).toBe(2)
  })

  it('computes headers correctly', () => {
    // Trigger computed property
    wrapper.setData({ filter: { start: '2024-01-01', end: '2024-01-31' } })

    const expectedTitle = 'Funcionários no período 2024-01-01 - 2024-01-31'
    expect(wrapper.vm.generalChartTitle).toBe(expectedTitle)
  })

  it('gets color by percentage correctly', () => {
    wrapper.setData({
      charts: [{ workload: 100 }]
    })

    expect(wrapper.vm.getColorByPercentage(80)).toBe('red')
    expect(wrapper.vm.getColorByPercentage(50)).toBe('yellow')
    expect(wrapper.vm.getColorByPercentage(20)).toBe('green')
  })
})