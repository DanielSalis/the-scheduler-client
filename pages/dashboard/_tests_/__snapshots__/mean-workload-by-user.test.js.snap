// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeanWorkloadByUser.vue renders the component and matches snapshot 1`] = `
"<gcontainer-stub class=\\"mean-workload-by-user\\">
  <v-row-stub tag=\\"div\\" align=\\"center\\" justify=\\"end\\" class=\\"mt-2\\" style=\\"width: 100%;\\">
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"2\\" tag=\\"div\\">
      <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" returnvalue=\\"2024-08-08\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" minwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"scale-transition\\">
        <v-date-picker-stub notitle=\\"true\\" width=\\"290\\" eventcolor=\\"warning\\" firstdayofweek=\\"0\\" localefirstdayofyear=\\"0\\" nexticon=\\"$next\\" nextmontharialabel=\\"$vuetify.datePicker.nextMonthAriaLabel\\" nextyeararialabel=\\"$vuetify.datePicker.nextYearAriaLabel\\" previcon=\\"$prev\\" prevmontharialabel=\\"$vuetify.datePicker.prevMonthAriaLabel\\" prevyeararialabel=\\"$vuetify.datePicker.prevYearAriaLabel\\" scrollable=\\"true\\" showcurrent=\\"true\\" selecteditemstext=\\"$vuetify.datePicker.itemsSelected\\" type=\\"date\\" value=\\"2024-08-08\\">
          <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
          <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" text=\\"true\\" type=\\"button\\">
            Cancel
          </v-btn-stub>
          <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" text=\\"true\\" type=\\"button\\">
            OK
          </v-btn-stub>
        </v-date-picker-stub>
      </v-menu-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"6\\" md=\\"2\\" tag=\\"div\\">
      <v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" openonclick=\\"true\\" contentclass=\\"\\" maxwidth=\\"auto\\" minwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" closeonclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"scale-transition\\">
        <v-date-picker-stub notitle=\\"true\\" width=\\"290\\" eventcolor=\\"warning\\" firstdayofweek=\\"0\\" localefirstdayofyear=\\"0\\" nexticon=\\"$next\\" nextmontharialabel=\\"$vuetify.datePicker.nextMonthAriaLabel\\" nextyeararialabel=\\"$vuetify.datePicker.nextYearAriaLabel\\" previcon=\\"$prev\\" prevmontharialabel=\\"$vuetify.datePicker.prevMonthAriaLabel\\" prevyeararialabel=\\"$vuetify.datePicker.prevYearAriaLabel\\" scrollable=\\"true\\" showcurrent=\\"true\\" selecteditemstext=\\"$vuetify.datePicker.itemsSelected\\" type=\\"date\\">
          <v-spacer-stub tag=\\"div\\"></v-spacer-stub>
          <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" text=\\"true\\" type=\\"button\\">
            Cancel
          </v-btn-stub>
          <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" text=\\"true\\" type=\\"button\\">
            OK
          </v-btn-stub>
        </v-date-picker-stub>
      </v-menu-stub>
    </v-col-stub>
    <v-col-stub cols=\\"12\\" sm=\\"12\\" md=\\"1\\" tag=\\"div\\" style=\\"text-align: center;\\">
      <div>
        <v-btn-stub color=\\"primary\\" dark=\\"true\\" tag=\\"button\\" activeclass=\\"\\" large=\\"true\\" block=\\"true\\" type=\\"button\\" center=\\"\\">
          Buscar
        </v-btn-stub>
      </div>
    </v-col-stub>
  </v-row-stub>
  <div>
    <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" class=\\"px-2 py-2\\" style=\\"height: 560px;\\">
      <div type=\\"bar\\" height=\\"100%\\" series=\\"[object Object]\\" options=\\"[object Object]\\"></div>
    </v-card-stub>
    <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" class=\\"mt-8 px-4\\" style=\\"width: 100%;\\">
      <v-row-stub tag=\\"div\\">
        <v-col-stub tag=\\"div\\">
          <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" value=\\"\\" backgroundcolor=\\"\\" label=\\"Filtre usuários (por nome)\\" prependicon=\\"mdi-magnify\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\"></v-text-field-stub>
        </v-col-stub>
      </v-row-stub>
    </v-card-stub>
    <v-expansion-panels-stub valuecomparator=\\"[Function]\\" value=\\"\\" activeclass=\\"v-item--active\\" multiple=\\"true\\" tag=\\"div\\" class=\\"mb-8\\">
      <v-expansion-panel-stub flat=\\"\\">
        <v-expansion-panel-header-stub expandicon=\\"$expand\\" class=\\"text-h6 ml-3 mean-workload-by-user__user-title\\">
          <v-row-stub tag=\\"div\\" style=\\"display: flex; flex-direction: row; justify-content: space-between;\\">
            <h6 style=\\"width: 33%;\\">
              Milena Matos
            </h6>
            <h6 style=\\"width: 33%;\\">
              milena@email.com
            </h6>
            <div class=\\"mr-4\\">
              <h6>
                165 minutos
              </h6>
              <v-chip-stub color=\\"red\\" activeclass=\\"\\" tag=\\"span\\" active=\\"true\\" closeicon=\\"$delete\\" closelabel=\\"$vuetify.close\\" filtericon=\\"$complete\\" style=\\"width: 100%; height: 5px;\\"></v-chip-stub>
            </div>
          </v-row-stub>
        </v-expansion-panel-header-stub>
        <v-expansion-panel-content-stub>
          <div type=\\"line\\" height=\\"550\\" options=\\"[object Object]\\" series=\\"[object Object]\\"></div>
        </v-expansion-panel-content-stub>
      </v-expansion-panel-stub>
    </v-expansion-panels-stub>
    <div class=\\"my-4\\">
      <div style=\\"display: flex; align-items: center;\\">
        <v-chip-stub color=\\"red\\" activeclass=\\"\\" tag=\\"span\\" active=\\"true\\" closeicon=\\"$delete\\" closelabel=\\"$vuetify.close\\" filtericon=\\"$complete\\" class=\\"mr-2\\" style=\\"width: 30px; height: 12px;\\"></v-chip-stub>
        <h4>
          Até 75% do valor mais alto
        </h4>
      </div>
      <div style=\\"display: flex; align-items: center;\\">
        <v-chip-stub color=\\"yellow\\" activeclass=\\"\\" tag=\\"span\\" active=\\"true\\" closeicon=\\"$delete\\" closelabel=\\"$vuetify.close\\" filtericon=\\"$complete\\" class=\\"mr-2\\" style=\\"width: 30px; height: 12px;\\"></v-chip-stub>
        <h4>
          De 25% a 75% comparado com o valor mais alto
        </h4>
      </div>
      <div style=\\"display: flex; align-items: center;\\">
        <v-chip-stub color=\\"green\\" activeclass=\\"\\" tag=\\"span\\" active=\\"true\\" closeicon=\\"$delete\\" closelabel=\\"$vuetify.close\\" filtericon=\\"$complete\\" class=\\"mr-2\\" style=\\"width: 30px; height: 12px;\\"></v-chip-stub>
        <h4>
          Menos que 25% do valor mais alto
        </h4>
      </div>
    </div>
  </div>
</gcontainer-stub>"
`;
